src/auctionHouse/index.ts

```diff
@@ -0,0 +1,71 @@
+import { Connection, PublicKey } from "@solana/web3.js";
+import { AuctionHouseSettings } from "./settings";
+import { AuctionHouseStats } from "./stats";
+
+export class AuctionHouseClient {
+    private connection: Connection;
+
+    constructor(connection: Connection) {
+        this.connection = connection;
+    }
+
+    loadAuctionHouse(auctionHouseKey: PublicKey): Promise<AuctionHouse> {
+        return new AuctionHouse({}).refresh();
+    }
+
+    createAuctionHouse(settings: AuctionHouseSettings): Promise<AuctionHouse> {
+        return new AuctionHouse({}).refresh();
+    }
+
+    updateAuctionHouse(auctionHouseKey: PublicKey, settings: AuctionHouseSettings): Promise<AuctionHouse> {
+        return new AuctionHouse({}).refresh();
+    }
+}
+
+export class AuctionHouse {
+    private settings: AuctionHouseSettings;
```

Comment:
I'd call those _args_ or _config_. It's not exactly _settings_ that are stored in a file or something.